### PR TITLE
fix issue #29: scan the jar, custom parsing service provider file to bypass the classloading issue on INDEX.LIST

### PR DESCRIPTION
**Pull request by missedone**
2nd try on issue #29

### CODE REVIEW

File: remote/src/main/java/org/testng/remote/support/ServiceLoaderHelper.java [hunk 1]
```diff
@@ -26,4 +33,118 @@ public static RemoteTestNGFactory getFirst(Version version) {
         }
         return factories.get(0);
     }
+
+	public static RemoteTestNGFactory getFirstQuietly(String version) {
+		return getFirstQuietly(Version.parseVersion(version));
+	}
+
+	public static RemoteTestNGFactory getFirstQuietly(Version version) {
```
Comment (missedone) [inline remote/src/main/java/org/testng/remote/support/ServiceLoaderHelper.java]: load the service provider file by scanning the jar file, and manually parse it

**Comment by missedone** [comment]
@juherr 
please coin in your thoughts.
thanks

File: remote/src/main/java/org/testng/remote/support/ServiceLoaderHelper.java [hunk 2]
```diff
@@ -26,4 +33,99 @@ public static RemoteTestNGFactory getFirst(Version version) {
...
+      }
+    }
+
+    throw new TestNGException(version + " is not a supported TestNG version");
+  }
+
+  // Parse a single line from the given configuration file, adding the name
+  // on the line to the names list.
+  //
+  private static int parseLine(Class service, URL u, BufferedReader r, int lc, List<String> names)
```
Comment (juherr) [inline remote/src/main/java/org/testng/remote/support/ServiceLoaderHelper.java]: Maybe `Class<?>` instead

File: remote/src/main/java/org/testng/remote/support/ServiceLoaderHelper.java [hunk 3]
```diff
@@ -26,4 +33,99 @@ public static RemoteTestNGFactory getFirst(Version version) {
...
+      throws IOException, ServiceConfigurationError {
+    String ln = r.readLine();
+    if (ln == null) {
+      return -1;
+    }
+    int ci = ln.indexOf('#');
+    if (ci >= 0)
```
Comment (juherr) [inline remote/src/main/java/org/testng/remote/support/ServiceLoaderHelper.java]: Brackets are missing here and below.

**Comment by juherr** [comment]
LGTM :+1:

**Pull request merged**